hudi-common/src/main/java/org/apache/hudi/common/table/log/AbstractHoodieLogRecordReader.java

```diff
@@ -217,16 +215,21 @@ public synchronized void scan(Option<List<String>> keys) {
             && !HoodieTimeline.compareTimestamps(logBlock.getLogBlockHeader().get(INSTANT_TIME), HoodieTimeline.LESSER_THAN_OR_EQUALS, this.latestInstantTime
         )) {
           // hit a block with instant time greater than should be processed, stop processing further
+          LOG.info("hit a block with instant time greater than should be processed, stop processing further. logfile: + " + logFile
+                  + " , blockType: " + logBlock.getBlockType() + " , instantTime: " + instantTime + " , latestInstantTime : " + latestInstantTime  );
           break;
         }
         if (logBlock.getBlockType() != CORRUPT_BLOCK && logBlock.getBlockType() != COMMAND_BLOCK) {
-          if (!completedInstantsTimeline.containsOrBeforeTimelineStarts(instantTime)
-              || inflightInstantsTimeline.containsInstant(instantTime)) {
+            if (!checkIfValidCommit(instantTime) || inflightInstantsTimeline.containsInstant(instantTime)) {
```

Comment:
what exactly are we trying to fix here. can you help explain the scenario.

Reply:
im explained in jira ：https://issues.apache.org/jira/browse/HUDI-3644

when log file block append sucess.  And deltacommit rollback. And this instance time is not before activeTimeline starts. This log file block will be merged, cause data duplication.

Reply:
sorry,i lost some code when commit, update now

Reply:
> And this instance time is before activeTimeline starts

How does that happen, is it archived ?